pkg/serverless/daemon/daemon.go

```diff
@@ -71,6 +73,51 @@ type Daemon struct {
 	// finishInvocationOnce assert that FinishedInvocation will be called only once (at the end of the function OR after a timeout)
 	// this should be reset before each invocation
 	finishInvocationOnce sync.Once
+
+	// metricsFlushMutex ensures that only one metrics flush can be underway at a given time
+	metricsFlushMutex sync.Mutex
+
+	// tracesFlushMutex ensures that only one traces flush can be underway at a given time
+	tracesFlushMutex sync.Mutex
+
+	// logsFlushMutex ensures that only one logs flush can be underway at a given time
+	logsFlushMutex sync.Mutex
+}
```

Comment:
maybe we could move those 3 mutexes into one object?

Reply:
Are you suggesting having 1 mutex instead of 3? Or somehow having one object with the 3 mutexes?

Reply:
one object with 3 mutexes

Reply:
What would the benefit of that change be? I think it's a bit simpler this way.